### ISSUE TITLE
choosing the best_model automatically for finalizing

### ISSUE THREAD

**Issue by sathishkumar369** [pycaret/pycaret#648]
Hi pycaret team!

I started using pycaret library since April. lot of request has been considered and solved. tons of thanks for your hard work and to Moez.

I have concern on selecting the best model

# compare specific models
best_specific = compare_models(whitelist = ['dt','rf','xgboost'])

In the above code I have passed 3 models for training. and it will give the best model on the top, based on the accuracy.

I'm expecting one more feature which is selecting the best model for training. so that we can finalize the model instead of giving custom model.
best_specific = compare_models(whitelist = ['dt','rf','xgboost'])
best_model_white_list = best_model(best_specific, n_select=1, sort='accuracy')
finalize_model(best_model_white_list)

Thanks

**Comment by Yard1** [comment]
`automl()` function returns the best model created so far.

**Comment by pycaret** [comment]
@sathishkumar369 What you are looking for is `automl` function but also `compare_models` returns the best model as well. In scenarios, when you are only using compare_models you can try something like this:

```
best = compare_models()
final_best = finalize_model(best)
```

If you are going beyond compare models and trying different models, tuning, ensembling etc. you can look at this example:

```
top3 = compare_models(n_select = 3)
tuned_top3 = [tune_model(i) for i in top3]
blender = blend_models(top3)
stacker = stack_models(top3)
best_model = automl()
```

**Comment by gm039** [comment]
Hi,

I am finding difficulty using automl.
Let's assume a case where we have only one model (QDA) created and fine-tuned. I am calling automl after it. 

Trying to predict on both (Tuned model and automl best model). But there is a difference in performance. Since the parameters of both the models are the same and hence the performance on holdout should be also the same. Can you please explain this? 

![InkedCapture_LI](https://user-images.githubusercontent.com/50217085/94714424-1fa6f300-031a-11eb-8bf1-114ff95e03d9.jpg)

**Comment by pycaret** [comment]
@gm039 This is because when you call `automl` function, the model is refitted on the entire dataset. Whereas the model generated with `tune_model` as you are comparing with is only fitted on the train set.
